File: src/main/java/impl/ConsoleSink.java
package impl;

import entities.LogLevel;
import entities.LogMessage;
import entities.Sink;

public class ConsoleSink implements Sink {
    LogLevel cutoffLogLevel;

    public ConsoleSink(LogLevel cutoffLogLevel){
        this.cutoffLogLevel = cutoffLogLevel;
    }
    @Override
    public void write(LogMessage message) {
        var logLevelComparison = message.level.compareTo(cutoffLogLevel);
        if (logLevelComparison >= 0){
            System.out.println(message.getLog());
        }
    }

    @Override
    public void clear() {
        System.out.print("\033[H\033[2J");
        System.out.flush();
    }
}

File: src/main/kotlin/impl/ConsoleSink.kt
package impl

import entities.LogLevel
import entities.LogMessage
import entities.Sink

class ConsoleSink(private var cutoffLogLevel: LogLevel) : Sink {
    override fun write(message: LogMessage?) {
        val logLevelComparison = message?.level?.compareTo(cutoffLogLevel)
        if (logLevelComparison != null) {
            if (logLevelComparison >= 0) {
                println(message.log)
            }
        }
    }

    override fun clear() {
        print("\u001b[H\u001b[2J")
        System.out.flush()
    }
}
